doc/tutorials/viz/histo3D/histo3D.markdown

```diff
@@ -0,0 +1,157 @@
+Creating a 3D histogram {#tutorial_histo3D}
+================
+
+Goal
+----
+
+In this tutorial you will learn how to
+
+-   Create your own callback keyboard function for viz window.
+-   Show your 3D histogram in a viz window.
+
+Code
+----
+
+You can download the code from [here ](https://github.com/opencv/opencv/tree/master/samples/cpp/tutorial_code/viz/histo3D.cpp).
+@include samples/cpp/tutorial_code/viz/histo3D.cpp
+
+Explanation
+-----------
+
+Here is the general structure of the program:
+
+-   You can give full path to an image in command line
+    @code{.cpp}
```

Comment:
You already have this code in `histo3D.cpp`. Please take a look at `how_to_use_OpenCV_parallel_for_.markdown` tutorial and the corresponding .cpp file to see how snippets can be taken form the source code.